crates/network-scanner-net/src/async_io.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    num::NonZeroUsize,
+    os::windows::io::{AsRawSocket},
+    sync::Arc,
+};
+
+use polling::{Event, Events};
+use socket2::Socket;
+use tokio::sync::{Mutex};
+
+use crate::tokio_raw_socket::TokioRawSocket;
+
+#[derive(Debug)]
+pub struct AsyncIoRuntime {
+    poller: Arc<polling::Poller>,
+    socket_id: usize,
+    loop_handle: Option<std::thread::JoinHandle<()>>,
+    waiting_list: Arc<Mutex<Vec<SocketWaiter>>>,
+}
+
+#[derive(Debug)]
+struct SocketWaiter {
+    id: usize,
+    waker: std::task::Waker,
+}
```

Comment:
suggestion: Use `Arc<Mutex<HashMap<usize, std::task::Waker>>>`, which seems to be the appropriate data structure for all the operations here, and remove this struct.

Reply:
updated